### PR TITLE
SummaryState: add serializer support

### PR DESCRIPTION
**Pull request by akva2**

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this opm-simulators=4079 please

File: opm/input/eclipse/Schedule/SummaryState.hpp [hunk 1]
```diff
@@ -122,6 +125,39 @@ class SummaryState {
...
+
+        st.elapsed = 1.0;
+        st.values = {{"test1", 2.0}};
+        st.well_values = {{"test2", {{"test3", 3.0}}}};
+        st.m_wells = {"test4"};
+        st.well_names = {"test5"};
+        st.group_values = {{"test6", {{"test7", 4.0}}}},
+        st.m_groups = {"test7"};
+        st.group_names = {"test8"},
+        st.conn_values = {{"test9", {{"test10", {{5, 6.0}}}}}};
```
Comment (bska) [inline opm/input/eclipse/Schedule/SummaryState.hpp:156]: This is probably okay as a way of testing the serialization code, but please be aware that this setup does not honour the internal invariants of the data members.  In other words, the object returned from `serializeObject()` here is not a valid `SummaryState`.

Comment (akva2) [reply opm/input/eclipse/Schedule/SummaryState.hpp:156]: yeah, i'm aware, but as it's only for the purpose of testing serialization i don't consider it a problem.

Comment (bska) [reply opm/input/eclipse/Schedule/SummaryState.hpp:156]: > as it's only for the purpose of testing serialization i don't consider it a problem.

Right.  Let's hope no-one chooses to call the function outside of the testing code.

**Review by bska — APPROVED** [review]
I think I've stated earlier that I'd prefer if the `serializeObject()` member function were named differently&mdash;e.g. `serializationTestObject()`&mdash;but I'm not going to rehash that argument here.  This is good enough for testing purposes so I'll merge into master.

**Pull request merged**